core/src/main/java/com/sequenceiq/cloudbreak/repository/StackRepository.java

```diff
@@ -462,8 +462,9 @@ List<ResourceWithId> getAsAuthorizationResourcesByCrns(@Param("id") Long id, @Pa
             "FROM Stack s " +
             "WHERE s.environmentCrn = :envCrn " +
             "AND s.terminated IS NULL " +
-            "AND tunnel <> :latestTunnel")
-    int getNotUpgradedStackCount(@Param("envCrn") String envCrn, @Param("latestTunnel") Tunnel latestTunnel);
+            "AND (s.type <> 'TEMPLATE' OR s.type IS NULL) " +
```

Comment:
How can `type` be `NULL` and why is it okay? Even then it would seem safer to check for e.g. `AND (s.type = 'DATALAKE' OR s.type = 'WORKLOAD')` (or a similar construct using `IN`). (The fourth value, `LEGACY`, is not something to encounter in CDP I think.)

Reply:
No clue, I found multiple similar filterings in this repository.
Although you are right, I will make it explicit.